### ISSUE TITLE
Support conditional formatting

### ISSUE THREAD

**Issue by renkun-ken** [yihui/knitr#974]
Consider the following example:

```rconsole
> library(knitr)
> df <- data.frame(name = c("James", "Ken", "Jenny"), age = c(18, 20, 22))
> kable(df)

|name  | age|
|:-----|---:|
|James |  18|
|Ken   |  20|
|Jenny |  22|
```

Is there a recommended way to transform the output to a html table with conditional formatting like

```
|name | age|
|:----|---:|
|James|<span style="background-color:#ffffff;">18</span>|
|Ken  |<span style="background-color:#ffffaa;">20</span>|
|Jenny|<span style="background-color:#ffff85;">22</span>|
```

so that the background color of cells in column `age` gives a visual indicator of the magnitude of the cell value relative to other cells in the same column.

One other possible use case is to make red texts in all `age` cells whose value >= 20 so that the produced code is

```
|name | age|
|:----|---:|
|James|  18|
|Ken  |<span style="color:red;">20</span>|
|Jenny|<span style="color:red;">22</span>|
```

Basically, the feature may require that `kable` allow for some intermediate intervention by functions that transform the text to be put into the cell given its column name and its original value (e.g. numeric value).

**Comment by yihui** [comment]
That sounds a job more appropriate to the [DT](https://github.com/rstudio/DT) package instead of the `kable()` function. I do not want to make `kable()` too fancy.

**Comment by renkun-ken** [comment]
Making `kable()` support something like this indeed is too fancy but using DT for such static output seems to be an overkill. Is it possible for user functions to post-process the text of columns generated by `kable()` so that as long as a column is generated, callback functions should be called to allow for post-processing given the original column vector so that user can further customize the final text to be put into the cells as texts.

If a user wants to customize the output text to be colorful by the following transformation function

```r
colorize_age <- function(value, text) { 
  sprintf('<span style="color: %s">%s</span>', rgb(value/max(value), 0, 0), text) 
}
```

So the column texts can be transformed like 

```rconsole
> colorize_age(df$age, as.character(df$age))
[1] "<span style=\"color: #D10000\">18</span>" "<span style=\"color: #E80000\">20</span>"
[3] "<span style=\"color: #FF0000\">22</span>"
```

To allow this, an api of callback function list may be added like 

```r
kable(df, callback = list(age = colorize_age))
```

so that the function will be called after the column text is ready to add to the text table. The returned character vector is the real value added.

**Comment by yihui** [comment]
Thanks for the example! I'll think about it.

**Comment by renkun-ken** [comment]
Really appreciate if you consider the following use cases:

* Red ages for age < 18.
* Bold names for those whose age > 22.

A more general interface may accept transform functions like `transform(df, value, text)` where `df` is the input data.frame of `kable()` so that the function has access to other column vectors, `value` is the column vector being processed, `text` is the text of that column produced by original `kable()`.

I'm not sure if such interface meets the design standards of `knitr` but I'll really appreciate it if such functionality is allowed :) This simply makes the wrong/dangerous numbers in an annoying text-table stand out and catch immediate attention in my html report.

**Comment by renkun-ken** [comment]
Create a new package [formattable](https://github.com/renkun-ken/formattable) to support conditional formatting that works with knitr.

**Comment by yihui** [comment]
Thanks! That looks very interesting. I'll take a closer look tomorrow.
